src/devices/machine/pit8253.cpp

```diff
@@ -306,7 +332,16 @@ void pit_counter_device::set_output(int output)
 		m_output = output;
 		LOG2("set_output() timer %d: %s\n", m_index, output ? "low to high" : "high to low");
 
-		downcast<pit8253_device *>(owner())->m_out_handler[m_index](output);
+		if(m_index == 3)
+		{
+			// only possible with a PS/2 PIT.
+			downcast<ps2_pit_device *>(owner())->m_ch3_out_handler(output);
+		}
+		else
+		{
+			downcast<pit8253_device *>(owner())->m_out_handler[m_index](output);
+		}
+
```

Comment:
I don’t particularly like having this extra check here, or the downcast for that matter.  Is there a better way?  Also, you aren’t following the style of the file (`if (...)` vs `if(...)`).